documentation/src/main/doc/modules/kafka/examples/inbound/KafkaBatchRecordExample.java

```diff
@@ -0,0 +1,46 @@
+package inbound;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.TopicPartition;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+
+import io.smallrye.reactive.messaging.kafka.KafkaBatchRecords;
+import io.smallrye.reactive.messaging.kafka.KafkaRecord;
+import io.smallrye.reactive.messaging.kafka.api.IncomingKafkaRecordMetadata;
+
+@ApplicationScoped
+public class KafkaBatchRecordExample {
+
+    @SuppressWarnings("unchecked")
+    // tag::code[]
+    @Incoming("prices")
+    public CompletionStage<Void> consumeMessage(KafkaBatchRecords<String, Double> records) {
+        for (KafkaRecord<String, Double> record : records) {
+            IncomingKafkaRecordMetadata<String, Double> metadata = record.getMetadata(IncomingKafkaRecordMetadata.class).orElse(null);
+            if (metadata != null) {
+                int partition = metadata.getPartition();
+                long offset = metadata.getOffset();
+                ConsumerRecord<String, Double> consumerRecord = metadata.getRecord();
+            }
+        }
+        return CompletableFuture.completedFuture(null);
```

Comment:
This is interesting. For a single message, we would `return msg.ack()`. How do we propagate ack here? Does `KafkaBatchRecords` have an `ack` method?

Reply:
yes in this example the `return records.ack()` would be correct.